src/FSharpVSPowerTools.Logic.VS2015/ResolveUnopenedNamespaceSuggestedActionsSourceProvider.fs

```diff
@@ -0,0 +1,84 @@
+﻿namespace FSharpVSPowerTools.Logic.VS2015
+
+open System.ComponentModel.Composition
+open Microsoft.VisualStudio.Language.Intellisense
+open System
+open Microsoft.VisualStudio.Text
+open Microsoft.VisualStudio.Text.Editor
+open Microsoft.VisualStudio.Text.Operations
+open Microsoft.VisualStudio.Utilities
+open FSharpVSPowerTools.ProjectSystem
+open System.Threading.Tasks
+open FSharpVSPowerTools.Refactoring
+open Microsoft.VisualStudio.Shell
+open FSharpVSPowerTools
+
+[<Export(typeof<ISuggestedActionsSourceProvider>)>]
+[<Name "Resolve Unopened Namespaces Suggested Actions">]
+[<ContentType "F#">]
+[<TextViewRole(PredefinedTextViewRoles.Editable)>]
+type ResolveUnopenedNamespaceSuggestedActionsSourceProvider() =
+    [<Import; DefaultValue>]
+    val mutable FsharpVsLanguageService: VSLanguageService
+
+    [<Import; DefaultValue>]
+    val mutable TextDocumentFactoryService: ITextDocumentFactoryService
+
+    [<Import(typeof<SVsServiceProvider>); DefaultValue>]
+    val mutable ServiceProvider: IServiceProvider
+
+    [<Import; DefaultValue>]
+    val mutable UndoHistoryRegistry: ITextUndoHistoryRegistry
+
+    [<Import; DefaultValue>]
+    val mutable ProjectFactory: ProjectFactory
+
+    interface ISuggestedActionsSourceProvider with
+        member x.CreateSuggestedActionsSource(textView: ITextView, buffer: ITextBuffer): ISuggestedActionsSource = 
+            if textView.TextBuffer <> buffer then null
+            else
+                let generalOptions = Setting.getGeneralOptions x.ServiceProvider
+                if obj.ReferenceEquals (generalOptions, null) || not generalOptions.ResolveUnopenedNamespacesEnabled then null
```

Comment:
No, it's `LanguagePrimitives.PhysicalEquality`. I added `==` as an alias for `ReferenceEquals`.